src/ctx.cpp

```diff
@@ -1641,6 +1641,18 @@ llvm::Value *FunctionEmitContext::BinaryOperator(llvm::Instruction::BinaryOps in
     int arraySize = lArrayVectorWidth(type);
     if (arraySize == 0) {
         llvm::Instruction *bop = llvm::BinaryOperator::Create(inst, v0, v1, name, bblock);
+        if (llvm::isa<llvm::IntegerType>(v0->getType())) {
+            const auto elemTy = llvm::dyn_cast<llvm::IntegerType>(v0->getType());
+            if (elemTy->getSignBit() > 0)
```

Comment:
This does not look like correct approach. `llvm::Type` doesn't not hold signedness information about Integer Types.

See the implementation of `getSignBit()`:
```c++
/// Return a uint64_t with just the most significant bit set (the sign bit, if
/// the value is treated as a signed number).
uint64_t getSignBit() const {
    return 1ULL << (getBitWidth()-1);
}
```

The information about integer type sign needs to be passed from explicitly to `BinaryOperator()`

Reply:
Nice catch!